internal/acctest/configs.go

```diff
@@ -850,3 +853,86 @@ resource "aws_iam_role_policy_attachment" "secrets_manager_read_write_update" {
 }
 `, rName, model))
 }
+
+// ConfigRandomPassword returns the configuration for a data source that
+// describes a random password. The data source is named 'test'. Use
+// data.aws_secretsmanager_random_password.test.random_password to
+// reference the password value.
+//
+// The function accepts a variable number of string arguments in the format
+// "key=value". The following keys are supported:
+//   - password_length: The length of the password. Default is 20.
+//   - exclude_punctuation: Whether to exclude punctuation characters. Default is true.
+//   - exclude_characters: A string of characters to exclude from the password.
+//   - exclude_lowercase: Whether to exclude lowercase letters. Default is false.
+//   - exclude_numbers: Whether to exclude numbers. Default is false.
+//   - exclude_uppercase: Whether to exclude uppercase letters. Default is false.
+//   - include_space: Whether to include a space character. Default is false.
+//   - require_each_included_type: Whether to require at least one character from each included type. Default is false.
+//
+// Ensure that you use lifecycle.ignore_changes to prevent Terraform from
+// detecting changes to the password value, as it is generated randomly on each
+// plan/apply. For example:
+//
+//	lifecycle {
+//	  ignore_changes = [password]
+//	}
+func ConfigRandomPassword(overrides ...string) string {
+	// Default configuration values
+	config := map[string]string{
+		"password_length":     "20",
+		"exclude_punctuation": "true",
+	}
+
+	// Additional keys without defaults
+	optionalKeys := []string{
+		"exclude_characters",
+		"exclude_lowercase",
+		"exclude_numbers",
+		"exclude_uppercase",
+		"include_space",
+		"require_each_included_type",
+	}
+
+	// Parse overrides and update the config map
+	for _, override := range overrides {
+		parts := strings.SplitN(override, "=", 2)
+		if len(parts) == 2 {
+			key := strings.TrimSpace(parts[0])
+			value := strings.TrimSpace(parts[1])
+			config[key] = value
+		}
+	}
+
+	// Build the Terraform configuration string
+	var builder strings.Builder
+	builder.WriteString(`
+data "aws_secretsmanager_random_password" "test" {
```

Comment:
Did you consider using `ephemeral "aws_secretsmanager_random_password"`? I know this would tie the tests to a newer version of Terraform (I think there's a check for that in `terraform-plugin-testing`) but I think that would prevent the need for
```
  lifecycle {
    ignore_changes = [password]
  }
```.

Reply:
Good idea! I changed it to use ephemeral resources.